[CloudSync] Update CloudSync code as per new Design (#544)
lf-edge/edge-home-orchestration-go: docs/cloudsync_mqtt.md

```
curl --location --request GET 'http://<ip where edge-orchestration is running>:56001/api/v1/orchestration/cloudsyncmgr/getsubscribedata/<topicname>/<appid>'
```
# 6. DataStructures Used
### a. URLData 

> This is a hashmap of the type <string,[]string> to store the client ids (appids) mapped to the Url of the broker to which the client are connected. For any first time entry of url in the hashmap, a connection is established between the broker and homeedge. For any subequent entry for the same url, the client id is appended to the list of client ids and same connection object is used for communication. In case of any client calling disconnect or wants to terminate the connection with broker, the client id can be removed from the list of client ids. Incase of one single client for any url calling the disconnect or wanting to terminate the connection, then the connection with the broker is closed and url entry with its key is removed from the Hashmap.

Reviewer: ```suggestion
This is a hashmap of the type <string,[]string> to store the client ids (appids) mapped to the Url of the broker to which the client are connected. For any first time entry of url in the hashmap, a connection is established between the broker and Home Edge. For any subsequent entry for the same url, the client id is appended to the list of client ids and same connection object is used for communication. In case of any client calling disconnect or wants to terminate the connection with broker, the client id can be removed from the list of client ids. In case of one single client for any url calling the disconnect or wanting to terminate the connection, then the connection with the broker is closed and url entry with its key is removed from the Hashmap.
```

### b. MQTTClient

> This hashmap of type<string,*Client> stores the client object for every new url connection between home edge and Broker.When a request for any url comes to Homeedge , this hashmap is searched to see if the client object exits. If the client object exist, the object is returned and further operations of publish/subscribe is performed. Incase if the client object is not present for the given url, then entry to MQTTClient hashmap is created with the url and object created. When all the clients for any url have closed the connection, entry from MQTTClient for that url is also removed.

Reviewer: ```suggestion
This hashmap of type<string,*Client> stores the client object for every new url connection between home edge and Broker. When a request for any url comes to Home Edge , this hashmap is searched to see if the client object exits. If the client object exist, the object is returned and further operations of publish/subscribe is performed. Incase if the client object is not present for the given url, then entry to MQTTClient hashmap is created with the url and object created. When all the clients for any url have closed the connection, entry from MQTTClient for that url is also removed.
```

### c. SubscriptionInfo

> This hashmap of type<{topic,url},[]clientid> stores the subscriptionInfo. The key used here is composite and uses both topic and url to uniquely identify the subscription. Multiple clients can subscribe to the same broker on same topic and so the list of such clientids is stored as value.Before retriveing the data requested by any client, this hashmap will be checked to see if the client is subscribed or not.

Reviewer: ```suggestion
This hashmap of type<{topic,url},[]clientid> stores the subscriptionInfo. The key used here is composite and uses both topic and url to uniquely identify the subscription. Multiple clients can subscribe to the same broker on same topic and so the list of such clientids is stored as value. Before retrieving the data requested by any client, this hashmap will be checked to see if the client is subscribed or not.
```

### d. SubscribedData

> This hashmap of type<{topic,url},string> stores the data published by any client on topic that has been subscribed by any client for the same broker.When a client requests for subsrciption to any topic, a entry is made to this hashmap with {topic,url} as the composite key and the latest published data as the string.Whenever a new data for the same key is published, the entry is updated. At any given point, this hashmap contains the latest value published for the topic on the broker url mentioned.

Reviewer: ```suggestion
This hashmap of type<{topic,url},string> stores the data published by any client on topic that has been subscribed by any client for the same broker. When a client requests for subscription to any topic, a entry is made to this hashmap with {topic,url} as the composite key and the latest published data as the string. Whenever a new data for the same key is published, the entry is updated. At any given point, this hashmap contains the latest value published for the topic on the broker url mentioned.
```

Author: Thanks for review comments!! Done with all changes in documentation!
